### PR TITLE
Faster tests with more asserts

### PR DESCRIPTION
**Pull request by kristjaneerik**
Here I improve the test suite a bit by making sure all plots are closed so they don't accumulate, make many of the tests faster by using a "thinned" dataset, consolidate some of the code with class inheritance, add assertions for many tests on Views.

The consolidation of test code seemed to make sense and enabled adding many assertions without much duplicated work. There's still flexibility, as each test class can override any test from the "parent" class, which I still do extensively for e.g. the violinplot view.

In the process I also added one failing test, because it seems like `set_default_scale` is broken. I've tried to fix it in a different branch I'm working on, but I haven't quite figured it out.

I also set the y label for 1D Views, like "Count" for a non-density based histogram and added assertions to check that -- previously, the x label was set to the channel even when the orientation was flipped, which was a bug, so in fixing that, I made this change.

I haven't added asserts for all tests related to Views, but I added TODOs for places that could use them... Can remove them if you'd like.

On this branch:
```
$ time nose2 -c packaging/nose2.cfg -s cytoflow/tests -N 12 -v
...
Ran 384 tests in 52.803s

FAILED (failures=1)

real    0m54.309s
user    8m31.747s
sys     0m11.036s
```

vs on master:

```
$ git checkout master
$ time nose2 -c packaging/nose2.cfg -s cytoflow/tests -N 12 -v
...
Ran 378 tests in 307.552s

OK

real    5m9.081s
user    27m48.691s
sys     0m22.108s
```

Committing the changes in `Yeast Dose Response.ipynb` was an accident, but there isn't anything actually changing there. That's the notebook where `set_default_scale` is used but no longer works as expected, which is how I saw the bug.

In terms of the GUI tests, they're still slow (run in ~15 mins for me), and still have some failing tests that I mentioned in #269.

### CODE REVIEW

**Comment by kristjaneerik** [comment]
I've merged in `master` to this branch.

There's still a broken test due to the bug in `set_default_scale` as mentioned in the PR text.

File: cytoflow/tests/test_channel_stat.py [hunk 1]
```diff
@@ -74,9 +73,11 @@ def testSubset(self):
                                      subset = "Dox == 10.0",
                                      function = len).apply(self.ex)
         stat = ex.statistics[("ByDox", "len")]
-       
-        self.assertEqual(stat.loc[False], 24801)
```
Comment (kristjaneerik) [inline cytoflow/tests/test_channel_stat.py:78]: Verified that these are incorrect by direct comparison to a `groupby` in pandas; I've added that code to the PR.
```
T      Dox
False  1.0     9953
       10.0    5601
True   1.0       47
       10.0    4399
dtype: int64
```

Note that when I run these new tests, they pass.

**Comment by bpteague** [comment]
Delete lots of code AND a massive speed-up? Yes please. Cleaning up this mess was one of those "when I get around to it" things that I never got around to. Obviously. And `setUpClass` is a neat trick - you clearly know `unittest` better than I do (:

File: cytoflow/tests/test_channel_stat.py (hunk 1 above)
Comment (bpteague) [reply cytoflow/tests/test_channel_stat.py:78]: These change a little bit each time I upgrade `statsmodels`. Probably should use some fuzzier matching...

**Comment by bpteague** [comment]
I found the issue with `set_default_scale` -- `xscale` and `yscale` are instances of a custom trait, an `Enum` whose only valid values are the scales that have been registered. When I was fixing it for a new version of `traits`, I seem to have commented out the function that sets the default (debugging or something) and never uncommented it. Good to go now.

**Comment by kristjaneerik** [comment]
Great!

**Pull request merged**